src/scripts/settings/userSettings.js

```diff
@@ -521,13 +539,17 @@ export class UserSettings {
      * @return {Query} Query.
      */
     loadQuerySettings(key, query) {
-        let values = this.get(key);
-        if (values) {
-            values = JSON.parse(values);
-            return Object.assign(query, values);
+        let sortSettings = this.get(key);
+        let filterSettings = this.get(key + filterSettingsPostfix, false);
+
+        if (sortSettings) {
+            sortSettings = filterQuerySettings(JSON.parse(sortSettings), allowedSortSettings);
+        }
+        if (filterSettings) {
+            filterSettings = filterQuerySettings(JSON.parse(filterSettings), allowedFilterSettings);
         }
 
-        return query;
+        return Object.assign(query, sortSettings, filterSettings);
```

Comment:
Sonarcloud marks `Object.assign` as a code smell. We should use the spread operator instead.

Reply:
:+1: good call